src/bitter/field_set.hpp

```diff
@@ -4,35 +4,38 @@
 // Distributed under the "BSD License". See the accompanying LICENSE.rst file.
 #pragma once
 
+#include <cassert>
+#include <cmath>
+#include <cstdint>
+
 #include "field_mask.hpp"
 #include "field_offset.hpp"
-
-#include <cstdint>
-#include <cassert>
+#include "field_size_in_bits.hpp"
 
 namespace bitter
 {
-/// @brief set the value of a field based on the index
-/// @param value is the data of the writer
-/// @param field is the data we want to write to value
+/// @brief set the value in a bitfield based on the index
+/// @param bitfield is the data of the writer
+/// @param value is the data we want to write to the bitfield
 template<class DataType, uint32_t Index, uint32_t... Sizes>
-DataType field_set(DataType value, DataType field)
+DataType field_set(DataType bitfield, DataType value)
 {
     uint32_t offset = field_offset<Index, Sizes...>();
     DataType mask = field_mask<DataType, Index, Sizes...>();
 
-    // Sanity check that we don't have garbage bits in our field
-    assert((field & (~mask)) == 0U);
+    // Verify that the bitfield can be represented with the available bits:
+    assert(value < std::pow(2, field_size_in_bits<Index, Sizes...>()) &&
```

Comment:
the old assert is however also very confusing for the user of the lib as there is no way (from the assertion) to deduce that the value you put into the function was too big.
If it is a matter of execution time, I can change the code to ensure the std pow statement is computed at compile-time.